dns.go

```diff
@@ -90,7 +125,7 @@ func (api *API) CreateDNSRecord(ctx context.Context, zoneID string, rr DNSRecord
 // This takes a DNSRecord to allow filtering of the results returned.
 //
 // API reference: https://api.cloudflare.com/#dns-records-for-a-zone-list-dns-records
-func (api *API) DNSRecords(ctx context.Context, zoneID string, rr DNSRecord) ([]DNSRecord, error) {
+func (api *API) DNSRecords(ctx context.Context, zoneID string, rr DNSRecord, params DNSListParameters) ([]DNSRecord, *ResultInfo, error) {
```

Comment:
if we're introducing a breaking change, let's update the method signature to use the `ResourceContainer` and be renamed to `ListDNSRecords`.

Reply:
Do you want me to do that for all the other methods as well?

Reply:
sure, if you're up for it!